google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelectionDialog.java

```diff
@@ -33,6 +103,346 @@
    */
   @Nullable
   CloudProject showDialog(@Nullable CloudProject cloudProject) {
-    return cloudProject;
+    // create dialog wrapper for the form once, create services and subscribe to login changes once.
+    if (dialogWrapper == null) {
+      dialogWrapper = new ProjectSelectionDialogWrapper();
+
+      dialogWrapper.setTitle(GctBundle.getString("cloud.project.selector.dialog.title"));
+      // disabled unless project is selected in the list.
+      dialogWrapper.setOKActionEnabled(false);
+      // install additional IDEA UI.
+      installTableSpeedSearch(projectListTable);
+
+      Stream.of(ProjectManager.getInstance().getOpenProjects())
+          .forEach(
+              project ->
+                  project
+                      .getMessageBus()
+                      .connect(dialogWrapper.getDisposable() /* disconnect once dialog is gone. */)
+                      .subscribe(
+                          GoogleLoginListener.GOOGLE_LOGIN_LISTENER_TOPIC,
+                          this::loadUsersAndProjects));
+    }
+
+    loadUsersAndProjects();
```

Comment:
I know we can go back and forth forever on naming, but just thinking after looking again..how about:

loadUsersAndProjects -> loadAllProjects
setCloudProject -> setActiveProject (I think the "cloud" part is implied enough)

Reply:
actually not setActiveProject, but setSelectedProject

Reply:
sounds good, and I'll change the getter name to reflect that idea as well - i.e. `getSelectedProject`